public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -93,15 +85,14 @@ export class DashboardGrid extends PureComponent<Props, State> {
 
   onLayoutChange = (newLayout: ReactGridLayout.Layout[]) => {
     for (const newPos of newLayout) {
-      this.panelMap[newPos.i!].updateGridPos(newPos, this.state.isLayoutInitialized);
+      this.panelMap[newPos.i!].updateGridPos(newPos, this.isLayoutInitialized);
     }
 
-    this.props.dashboard.sortPanelsByGridPos();
-
-    // This is called on grid mount as it can correct invalid initial grid positions
-    if (!this.state.isLayoutInitialized) {
-      this.setState({ isLayoutInitialized: true });
+    if (this.isLayoutInitialized) {
+      this.isLayoutInitialized = true;
     }
```

Comment:
There is no need to store isLayoutInitialized in state, we do not need to trigger a re-render here. Just need to remember that layout is initialized to distinguish later calls to onLayoutChange as "user initiated" vs the one that happens onmount